src/MiniExcel/SaveByTemplate/ExcelOpenXmlTemplate.cs

```diff
@@ -0,0 +1,126 @@
+
+        static ExcelOpenXmlTemplate()
+        {
+            _isExpressionRegex = new Regex("(?<={{).*?(?=}})");
+            _ns = new XmlNamespaceManager(new NameTable());
+            _ns.AddNamespace("x", Config.SpreadsheetmlXmlns);
+            _ns.AddNamespace("x14ac", Config.SpreadsheetmlXml_x14ac);
+        }
+
+        private readonly Stream _stream;
+        private readonly OpenXmlConfiguration _configuration;
+        private readonly IInputValueExtractor _inputValueExtractor;
+        private readonly StringBuilder _calcChainContent = new StringBuilder();
+
+        public ExcelOpenXmlTemplate(Stream stream, IConfiguration configuration, InputValueExtractor inputValueExtractor)
+        {
+            _stream = stream;
+            _configuration = (OpenXmlConfiguration)configuration ?? OpenXmlConfiguration.DefaultConfig;
+            _inputValueExtractor = inputValueExtractor;
+        }
+
+        public void SaveAsByTemplate(string templatePath, object value)
+        {
+            using (var stream = FileHelper.OpenSharedRead(templatePath))
+                SaveAsByTemplateImpl(stream, value);
+        }
+
+        public void SaveAsByTemplate(byte[] templateBtyes, object value)
+        {
+            using (Stream stream = new MemoryStream(templateBtyes))
+                SaveAsByTemplateImpl(stream, value);
+        }
+
+        public void SaveAsByTemplateImpl(Stream templateStream, object value)
+        {
+            //only support xlsx
+            templateStream.CopyTo(_stream);
+
+            var reader = new ExcelOpenXmlSheetReader(_stream, null);
+            var _archive = new ExcelOpenXmlZip(_stream, mode: ZipArchiveMode.Update, true, Encoding.UTF8);
+            {
+                //read sharedString
+                var sharedStrings = reader._sharedStrings;
+                StringBuilder calcSheetContent = new StringBuilder();
+
+                //read all xlsx sheets
+                var sheets = _archive.zipFile.Entries.Where(w =>
+                    w.FullName.StartsWith("xl/worksheets/sheet", StringComparison.OrdinalIgnoreCase)
+                    || w.FullName.StartsWith("/xl/worksheets/sheet", StringComparison.OrdinalIgnoreCase)
+                ).ToList();
+
+                int sheetIdx = 0;
+                foreach (var sheet in sheets)
+                {
+                    this.XRowInfos =
+                        new List<XRowInfo>(); //every time need to use new XRowInfos or it'll cause duplicate problem: https://user-images.githubusercontent.com/12729184/115003101-0fcab700-9ed8-11eb-9151-ca4d7b86d59e.png
+                    this.XMergeCellInfos = new Dictionary<string, XMergeCell>();
+                    this.NewXMergeCellInfos = new List<XMergeCell>();
+
+                    var sheetStream = sheet.Open();
+                    var fullName = sheet.FullName;
+
+                    var inputValues = _inputValueExtractor.ToValueDictionary(value);

```

Comment:
This is where the new class is called: `InputValueExtractor.ToValueDictionary()`